src/atlas/parallel/HaloAdjointExchangeImpl.h

```diff
@@ -0,0 +1,101 @@
+/*
+ * (C) British Crown Copyright 2020 Met Office
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ */
```

Comment:
Just for clarification - does that mean that the Copyright needs to be ecmwf's? I presume so, but want to be 100% sure before sending an email to the appropriate MetOffice people.

Reply:
Indeed, that is correct.

Reply:
There has been a MetOffice ECMWF legal discussion on this, and it appears that it is o.k. to put Crown copyright on new files that are created by MetOffice staff.  It seems not to contravene the CLA agreement.

Reply:
Hello Marek,

My management is looking at legal issues related with people from Météo-France adding some code to Atlas. They claim that I am not allowed to sign the CLA (which I did), and that Météo-France has to do it so that I can contribute.
They would like to know what kind of agreement has been concluded between ECMWF and UKMO. Could you explain what this agreement is like ?

Regards,

Philippe Marguinaud

Reply:
To provide final update to this. It is OK to have the Copyright belonging to the British Crown, as long as the CLA is signed which means the contributions will follow the Apache-2 license